dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -50,613 +50,620 @@
+    }
+
+
+    public Folder findFolderByPath(String path, String hostid, User user, boolean respectFrontEndPermissions) throws DotStateException,
+            DotDataException, DotSecurityException {
+        if(user==null) user = APILocator.getUserAPI().getSystemUser();
+        Host host = APILocator.getHostAPI().find(hostid, user, false);
+        return findFolderByPath(path,host,user, respectFrontEndPermissions);
+    }
+
+    @WrapInTransaction
+    public boolean renameFolder(final Folder folder, final String newName,
+                                final User user, final boolean respectFrontEndPermissions) throws DotDataException,
+            DotSecurityException {
+
+        boolean renamed = false;
+
+        if (!permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndPermissions)) {
+            throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to edit folder" + folder.getPath());
+        }
+
+        try {
+
+            renamed = folderFactory.renameFolder(folder, newName, user, respectFrontEndPermissions);
+
+            return renamed;
+        } catch (Exception e) {
+
+            throw new DotDataException(e.getMessage(),e);
+        }
+    }
+
+    @CloseDBIfOpened
+    public Folder findParentFolder(final Treeable asset, final User user, final boolean respectFrontEndPermissions) throws DotIdentifierStateException,
+            DotDataException, DotSecurityException {
+        Identifier id = APILocator.getIdentifierAPI().find(asset.getIdentifier());
+        if(id==null) return null;
+        if(id.getParentPath()==null || ROOT_PATH.equals(id.getParentPath()) || SYSTEM_FOLDER_PATH.equals(id.getParentPath())){
+            return null;
+        }
+        Host host = APILocator.getHostAPI().find(id.getHostId(), user, respectFrontEndPermissions);
+        Folder f = folderFactory.findFolderByPath(id.getParentPath(), host);
+
+        if(f == null || !UtilMethods.isSet(f.getInode())) return null;
+        if (!permissionAPI.doesUserHavePermission(f, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
+            if(UtilMethods.isSet(f.getPath())){
+                //Folder exists in DB, but the user does not have permissions to read it.
+                Logger.error(this, "User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to read Folder " + f.getPath());
+            }else{
+                //Despite the Folder Object is not null, It may return an empty Folder Object because the Parent Folder is missing.
+                Logger.error(this, "User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to read Folder " + id.getParentPath() + " Please check the folder exists.");
+            }
+            throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to read Folder " + f.getPath());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 159 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)